yb-voyager/cmd/importData.go

```diff
@@ -915,54 +896,123 @@ func doOneImport(task *SplitFileImportTask, connPool *tgtdb.ConnectionPool) {
 	}
 }
 
-func executeSqlFile(file string) {
-	log.Infof("Execute SQL file %q on target %q", file, target.Host)
-	connectionURI := target.GetConnectionUri()
-	conn, err := pgx.Connect(context.Background(), connectionURI)
+func newTargetConn() *pgx.Conn {
+	conn, err := pgx.Connect(context.Background(), target.GetConnectionUri())
 	if err != nil {
 		utils.WaitChannel <- 1
 		<-utils.WaitChannel
 		utils.ErrExit("connect to target db: %s", err)
 	}
-	defer conn.Close(context.Background())
 
-	if sourceDBType != POSTGRESQL && target.Schema != YUGABYTEDB_DEFAULT_SCHEMA {
-		setSchemaQuery := fmt.Sprintf("SET SCHEMA '%s'", target.Schema)
-		_, err := conn.Exec(context.Background(), setSchemaQuery)
-		if err != nil {
-			utils.ErrExit("run query %q on target %q: %s", setSchemaQuery, target.Host, err)
-		}
+	setTargetSchema(conn)
+	return conn
+}
+
+func setTargetSchema(conn *pgx.Conn) {
+	if sourceDBType == POSTGRESQL || target.Schema == YUGABYTEDB_DEFAULT_SCHEMA {
+		// For PG, schema name is already included in the object name.
+		// No need to set schema if importing in the default schema.
+		return
 	}
 
+	setSchemaQuery := fmt.Sprintf("SET SCHEMA '%s'", target.Schema)
+	_, err := conn.Exec(context.Background(), setSchemaQuery)
+	if err != nil {
+		utils.ErrExit("run query %q on target %q: %s", setSchemaQuery, target.Host, err)
+	}
+}
+
+func dropIdx(conn *pgx.Conn, idxName string) {
+	dropIdxQuery := fmt.Sprintf("DROP INDEX IF EXISTS %s", idxName)
+	log.Infof("Dropping index: %q", dropIdxQuery)
+	_, err := conn.Exec(context.Background(), dropIdxQuery)
+	if err != nil {
+		utils.ErrExit("Failed in dropping index %s: %v", idxName, err)
+	}
+}
+
+func executeSqlFile(file string, objType string) {
+	log.Infof("Execute SQL file %q on target %q", file, target.Host)
+
+	conn := newTargetConn()
+	defer func() {
+		if conn != nil {
+			defer conn.Close(context.Background())
```

Comment:
You're already in the `defer` context!!! Drop the `defer` keyword from this statement.